src/main/java/org/marketplace/cache/TokenCache.java

```diff
@@ -0,0 +1,29 @@
+package org.marketplace.cache;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class TokenCache {
+    private final Map<String, String> tokenCache = new ConcurrentHashMap<>();
+    Logger logger = LoggerFactory.getLogger(TokenCache.class);
+
+    public void saveToken(String username, String token){
+        tokenCache.put(username, token);
+        logger.info("Token added to cache for user: " + username);
+    }
+
+    public String getToken(String username){
+        logger.info("Getting token for user: " + username);
+        return tokenCache.get(username);
+    }
+
+    public void removeToken(String username){
```

Comment:
we remove a token on logout, this if fine 👍
the token might get expired so that we keep adding then expired token, we can spin up some listener that will make some housekeeping in this regard and schedule some check of tokens, so if any gets expired we remove it from the cache, what do you think?